.github/workflows/scripts/tf_configure_remote_backend.sh

```diff
@@ -0,0 +1,27 @@
+#!/bin/bash
+
+# This file is used to configure the remote backend for Terraform
+# We keep local backend to align with the docs and for easy try out
+# This script will replace the local backend with the S3 backend
+# lastly export a variable to be used by Terraform to confgire the remote backend
+
+# Check if a file path is provided as an argument
+if [ $# -eq 0 ]; then
+    echo "Usage: $0 <file>"
+    exit 1
+fi
+
+# Check if the file exists
+if [ ! -f "$1" ]; then
+    echo "File $1 not found."
+    exit 1
+fi
+
+# if running on mac requires extra sed -i '' '...' $f1
```

Comment:
nice suggestion, i think it's fine for this case since it's solely for GHA anyway.
There are a couple of tools that work different on Mac, sometimes it makes sense to just install the GNU version.